FE/src/components/PriceChart/index.tsx

```diff
@@ -24,47 +24,55 @@ export default function PriceChart({ chartInfo, axis, xStep, yStep }: PriceChart
   const maximumX = Math.max(...xDataset);
   const maximumY = Math.max(...yDataset);
 
-  const [minXThumb, setMinXThumb] = useState(0);
-  const [maxXThumb, setMaxXThumb] = useState(maximumX);
+  const [leftThumbX, setLeftThumbX] = useState(0);
+  const [rightThumbX, setRightThumbX] = useState(maximumX);
 
   // FIXME: handleMaxPriceChange와 handleMinPriceChange 내부 로직이 비슷한 흐름인데 중복을 줄일 수 없을까?
   const handleMaxPriceChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     const {
       target: { value },
     } = event;
     const newValue = Number(value);
```

Comment:
```suggestion
const handleMovePrice = ({ event, thumbX, currentDirection }: HandleMovePriceProps) => {
    const {
      target: { value },
    } = event;
    const newValue = Number(value);

    const isOverThumbX = newValue <= thumbX;
    const isBelowThumbX = newValue >= thumbX;

    const conditions = [
      { direction: 'left', value: isBelowThumbX },
      { direction: 'right', value: isOverThumbX },
    ];

    const condition = conditions.find(({ direction }) => direction === currentDirection);

    if (condition?.value) {
      if (condition?.direction === 'right') {
        setLeftThumbX(newValue - xStep);
      } else if (condition?.direction === 'left') {
        setRightThumbX(newValue + xStep);
      }
    }

    if (condition?.direction === 'left') {
      setLeftThumbX(newValue);
    }
    if (condition?.direction === 'right') {
      setRightThumbX(newValue);
    }
  };
```

이러한 식으로 리팩토링하려다가 실패했습니다.
if문에 left, right인지 비교는 오류가 안나게 최대한 명확하게 작성하다보니 저렇게 되었습니다. 원래 처음 리팩토링할때는 없었지만...

[이것](https://www.notion.so/f540bf6e7a64417c9e05afb17e1d8321)이 시도한 전체 코드입니다.

Reply:
저도 이런식으로 하나의 함수로 합치려고 시도했었는데 어려웠어요 ㅠㅠ 이든이 고민하신 것처럼 하나의 함수로 합치려다가  함수 내부에서 if문으로 분기처리를 해야해서 복잡해진다는 느낌을 받았어요.